mmv1/products/vertexai/FeatureOnlineStoreFeatureview.yaml

```diff
@@ -118,3 +124,62 @@ properties:
         description: |
           Columns to construct entityId / row keys. Start by supporting 1 only.
         item_type: Api::Type::String
+  - !ruby/object:Api::Type::NestedObject
+    name: 'vectorSearchConfig'
+    description: |
+      Configuration for vector search. It contains the required configurations to create an index from source data, so that approximate nearest neighbor (a.k.a ANN) algorithms search can be performed during online serving.
+    immutable: true
+    min_version: beta
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'embeddingColumn'
+        required: true
+        description: |
+          Column of embedding. This column contains the source data to create index for vector search. embeddingColumn must be set when using vector search.
+      - !ruby/object:Api::Type::Array
+        name: 'filterColumns'
+        description: |
+          Columns of features that're used to filter vector search results.
+        item_type: Api::Type::String
+      - !ruby/object:Api::Type::String
+        name: 'crowdingColumn'
+        description: |
+          Column of crowding. This column contains crowding attribute which is a constraint on a neighbor list produced by nearest neighbor search requiring that no more than some value k' of the k neighbors returned have the same value of crowdingAttribute.
+      - !ruby/object:Api::Type::Enum
+        name: distanceMeasureType
+        description: |
+          The distance measure used in nearest neighbor search.
+          SQUARED_L2_DISTANCE: Euclidean (L_2) Distance.
+
+          COSINE_DISTANCE: Cosine Distance. Defined as 1 - cosine similarity.
+
+          DOT_PRODUCT_DISTANCE: Dot Product Distance. Defined as a negative of the dot product.
+        values:
+          - :SQUARED_L2_DISTANCE
+          - :COSINE_DISTANCE
+          - :DOT_PRODUCT_DISTANCE
+      - !ruby/object:Api::Type::NestedObject
+        name: 'treeAhConfig'
+        exactly_one_of:
+          - treeAhConfig
+          - bruteForceConfig
+        description: |
+          Configuration options for the tree-AH algorithm (Shallow tree + Asymmetric Hashing).
+        properties:
+          - !ruby/object:Api::Type::String
+            name: 'leafNodeEmbeddingCount'
+            description: |
+              Number of embeddings on each leaf node. The default value is 1000 if not set.
```

Comment:
Is this default value likely to change in the future? If it's likely to remain stable (or if updates could happen during yearly major releases) using [`default_value`](https://googlecloudplatform.github.io/magic-modules/develop/field-reference/#default_value) would be a better experience for users.

Reply:
Since it is a beta field and considering that this resource is a very recent one that is being actively worked on, I thought it'd be better to use default_from_api rather than setting a default_value; anticipating that it might be changed later on. Although, I don't have an insight into how likely that is. I'd think once this field is GA It can be updated to use default_value probably?

Reply:
changing from default_from_api to default_value is tricky and not always non-breaking. For example, if a user sets this field to, say, 500, then removes the field from their configuration, Terraform will continue to set it to 500 - since that's the last value returned by the API. Changing to `default_value: 1000` would alter the configuration without the user's input.

Leaving it as default_from_api is fine if you're not sure the value is stable.